### DISCUSSION TITLE
Failed to execute youtube-dl: exit status 1

### DISCUSSION THREAD

**Discussion by LucasJanin** [mxpv/podsync#556] [General]
Hi

Today, the download doesn't work anymore but it was working yesterday.

```
time="2023-02-17T16:49:14-05:00" level=info msg="download count: 1"
time="2023-02-17T16:49:14-05:00" level=info msg="! downloading episode https://youtube.com/watch?v=CjsPSHRuwIM" episode_id=CjsPSHRuwIM index=0
time="2023-02-17T16:49:26-05:00" level=error msg="youtube-dl error: /tmp/podsync-4005477621/CjsPSHRuwIM.%(ext)s" error="failed to execute youtube-dl: exit status 1"
time="2023-02-17T16:49:26-05:00" level=error msg="[youtube] CjsPSHRuwIM: Downloading webpage\n[youtube] CjsPSHRuwIM: Downloading android player API JSON\nERROR: [youtube] CjsPSHRuwIM: Unable to extract uploader id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U\n"
time="2023-02-17T16:49:27-05:00" level=info msg="downloaded 0 episode(s)"
```

Does anyone have the same issue?

Thanks

**Comment by LucasJanin** [comment]
It looks like related to https://github.com/yt-dlp/yt-dlp/issues/6247

I apply this fix in my docker container and the download is working back
https://github.com/yt-dlp/yt-dlp/issues/6247#issuecomment-1434551530

**Comment by berndschmidl** [comment]
I have the same problem but the fix didn't work for me. Does anyone have other ideas? I'm also not sure if I'm using yt-dlp or youtube-dl. Where can I look this up? Thanks

**Comment by Marx2** [comment]
I've used docker image from
https://hub.docker.com/r/tdeutsch/podsync
Build 4 days ago.
So far so good

**Comment by dazkarieh** [comment]
same issue in QNAP SSH, any sugguestion?

```
time="2023-02-20T11:52:14Z" level=info msg="! downloading episode https://youtube.com/watch?v=cx5cVuPjEDI" episode_id=cx5cVuPjEDI index=0
time="2023-02-20T11:52:17Z" level=error msg="youtube-dl error: /tmp/podsync-1022987373/cx5cVuPjEDI.%(ext)s" error="failed to execute youtube-dl: exit status 1"
time="2023-02-20T11:52:17Z" level=error msg="[youtube] cx5cVuPjEDI: Downloading webpage\n[youtube] cx5cVuPjEDI: Downloading android player API JSON\nERROR: [youtube] cx5cVuPjEDI: Unable to extract uploader id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U\n"
time="2023-02-20T11:52:17Z" level=info msg="! downloading episode https://youtube.com/watch?v=k28OY-pX7fU" episode_id=k28OY-pX7fU index=1
time="2023-02-20T11:52:20Z" level=error msg="youtube-dl error: /tmp/podsync-1913794801/k28OY-pX7fU.%(ext)s" error="failed to execute youtube-dl: exit status 1"
time="2023-02-20T11:52:20Z" level=error msg="[youtube] k28OY-pX7fU: Downloading webpage\n[youtube] k28OY-pX7fU: Downloading android player API JSON\nERROR: [youtube] k28OY-pX7fU: Unable to extract uploader id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U\n"
time="2023-02-20T11:52:20Z" level=info msg="! downloading episode https://youtube.com/watch?v=rLn8diElHOk" episode_id=rLn8diElHOk index=2
time="2023-02-20T11:52:22Z" level=error msg="youtube-dl error: /tmp/podsync-2596225763/rLn8diElHOk.%(ext)s" error="failed to execute youtube-dl: exit status 1"
time="2023-02-20T11:52:22Z" level=error msg="[youtube] rLn8diElHOk: Downloading webpage\n[youtube] rLn8diElHOk: Downloading android player API JSON\nERROR: [youtube] rLn8diElHOk: Unable to extract uploader id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U\n"
time="2023-02-20T11:52:22Z" level=info msg="downloaded 0 episode(s)"
```

**Comment by dazkarieh** [comment]
> I've used docker image from https://hub.docker.com/r/tdeutsch/podsync Build 4 days ago. So far so good

well done, it worked 4 me

**Comment by berndschmidl** [comment]
I don't use docker and my installation worked fine before this bug. I guess I would just need help to get youtube-dl or yt-dlp running again. Not a developer and more winging it here

**Comment by LucasJanin** [comment]
> I don't use docker and my installation worked fine before this bug. I guess I would just need help to get youtube-dl or yt-dlp running again. Not a developer and more winging it here

Have you tried this? https://github.com/yt-dlp/yt-dlp/issues/6247#issuecomment-1434551530

**Comment by berndschmidl** [comment]
yes. didn't work.

**Comment by streightedge** [comment]
> > I've used docker image from https://hub.docker.com/r/tdeutsch/podsync Build 4 days ago. So far so good
> 
> well done, it worked 4 me

Is this only working for arm? I tried to pull image to amd and got no manifest found error. Am I doing something wrong?

**Comment by elias-fox** [comment]
I was running into this as well. The problem is you need to update yt-dlp. Fortunately the container can handle this automatically by adding the following to your config file:
```
# Youtube-dl specific configuration.
[downloader]
# Optional, auto update youtube-dl every 24 hours
self_update = true
```

**Comment by streightedge** [comment]
A good to know thank you.

I have fixed it temporarily by executing in container
python3 -m pip install --force-reinstall https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz

but it works until the container crashes for any reason.

**Comment by guss77** [comment]
I also encountered this problem with current `mxpv/podsync:latest` from docker hub. Using `tdeutsch/podsync` instead, updating `yt-dlp` locally inside the container or just rebuilding the container using the original `Dockerfile` all workaround the issue.

I believe its just a bug in the 6 months old version of `yt-dlp` in the docker hub image (very likely just an API change in YouTube) that is already fixed, and @mxpv should just trigger a rebuild of the docker hub image to get this fixed.

**Comment by skofield05** [comment]
Still no luck for me. I get this error now

`error="could not find youtube-dl: failed to execute youtube-dl: exit status 127"`

maybe I edited something in the past. but where would I  tell podsync where it can find youtube-dl or yt-dlp? Wouldn't care which one it uses.

**Comment by mxpv** [comment]
Can you try recent release: https://github.com/mxpv/podsync/releases/tag/v2.6.1 ?
```
docker pull ghcr.io/mxpv/podsync:latest
```

**Comment by skofield05** [comment]
I finally figured it out. I don't do docker but I got the newest version from github. Then I made sure that neither youtube-dl nor yt-dlp are on the system with purge and also uninstalling it via pip. You can check if it is still on the system with `whereis youtube-dl` and `whereis yt-dlp` 
Then I downloaded newest yt-dlp version via `pip` and made a symlink from youtube-dl to yt-dlp. That did the trick. Make sure to get yt-dlp from pip and not apt install.

**Comment by jclendineng** [comment]
Same issue persists
